### PR TITLE
Richardcho/correlationid answeredby

### PR DESCRIPTION
**Pull request by richardcho-msft**
- Add correlation_id to call_connection_properties
- Add answered_by_identifier to call_connection_properties
- Add operation_context to answer_request

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py [hunk 1]
```diff
@@ -248,6 +250,8 @@ def __init__(
         source_caller_id_number: Optional[PhoneNumberIdentifier] = None,
         source_display_name: Optional[str] = None,
         source_identity: Optional[CommunicationIdentifier] = None,
+        correlation_id: Optional[str] = None,
```
Comment (minwoolee-msft) [inline sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py:253]: missing docstring

File: sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py [hunk 2]
```diff
@@ -127,6 +129,7 @@ def from_connection_string(
 
         return cls(endpoint, access_key, **kwargs)
 
+    @distributed_trace
```
Comment (minwoolee-msft) [inline sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py:132]: one more thing - is there reason to add this here?
from doc says : https://azure.github.io/azure-sdk/python_implementation.html#distributed-tracing

"DO create a new span for each outgoing network call. If using the HTTP pipeline, the new span is created for you."

as this is not network call (its just getting client), would this need distributed trace?

Comment (richardcho-msft) [reply sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py:132]: yea not sure. Pylint complained about it. Should we ignore it?

Comment (minwoolee-msft) [reply sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_call_automation_client_async.py:132]: checked with other ACS team - they seem to include distributed tracing for even non-network call methods. we can probably keep as is then.

**Review by minwoolee-msft — APPROVED** [review]

**Pull request merged**